addons/edi/models/edi_issues.py

```diff
@@ -34,54 +31,58 @@ class ProjectTask(models.Model):
+    _name = "main.issues"
+    _description = "Main Issue Object"
 
-    _name = 'edi.issues'
-    _description = "EDI Issue-Tracked Object"
-    _inherit = ['mail.thread']
+    # Empty list variable, will be overriden in the specific models where used.
+    ISSUES_FIELD_MAP = []
 
-    def _default_project_id(self):
-        return self.env.ref('edi.project_default')
-
-    project_id = fields.Many2one('project.project', string="Issue Tracker",
-                                 required=True, default=_default_project_id)
-    issue_ids = fields.One2many('project.task', string="Issues",
-                                domain=['|', ('stage_id.fold', '=', False),
-                                        ('stage_id', '=', False)])
-    issue_count = fields.Integer(string="Issue Count",
-                                 compute='_compute_issue_counts', store=True)
-    rel_issue_count = fields.Integer(string="Related Issue Count",
-                                     compute='_compute_issue_counts',
-                                     store=True)
+    issue_ids = fields.One2many(
+        "project.task",
+        string="Issues",
+        domain=["|", ("stage_id.fold", "=", False), ("stage_id", "=", False)],
+    )
+    issue_count = fields.Integer(string="Issue Count", compute="_compute_issue_counts", store=True)
+    rel_issue_count = fields.Integer(
+        string="Related Issue Count", compute="_compute_issue_counts", store=True
+    )
 
-    @api.multi
-    @api.depends('issue_ids', 'issue_ids.stage_id', 'issue_ids.edi_doc_id',
-                 'issue_ids.edi_gateway_id', 'issue_ids.edi_transfer_id')
+    @api.model
+    def display_issue_counts_depends(self):
+        """
+        Main fields which _compute_issue_counts depends on, it will be extended with super call
+        on specific models where used.
+        """
+        return ["issue_ids", "issue_ids.stage_id"]
+
+    @api.depends(lambda self: self.display_issue_counts_depends())
     def _compute_issue_counts(self):
         """Compute number of open issues (for UI display)"""
-        inverse = self._fields['issue_ids'].inverse_name
-        related = [edi_field
-                   for field, edi_field in EDI_FIELD_MAP
-                   if edi_field != inverse and field not in self._fields]
+        inverse = self._fields["issue_ids"].inverse_name
+        related = [
+            edi_field
+            for field, edi_field in self.ISSUES_FIELD_MAP
+            if edi_field != inverse and field not in self._fields
+        ]
         for issues in self:
-            issues.issue_count = len(issues.issue_ids.filtered(
-                lambda x: all(not getattr(x, f) for f in related)
-            ))
-            issues.rel_issue_count = (len(issues.issue_ids) -
-                                      issues.issue_count)
+            issues.issue_count = len(
+                issues.issue_ids.filtered(lambda x: all(not getattr(x, f) for f in related))
```

Comment:
Do you think `not any(getattr(x, f) for f in related)` might be clearer?  I find that `not` in the middle of the expression makes it harder to understand, but maybe that's just me.

Reply:
That was in edi, and tbh i tried to make as few changes on edi so to minimise the risk of affecting smth else.
At this point i would not change anything there. I am brave but not so much :)